File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> countMap = new HashMap<>();
        for(int num : nums)
            countMap.put(num,countMap.getOrDefault(num,0)+1);
        
        Queue<Integer> maxHeap = new PriorityQueue<>(
            (a,b)->countMap.get(a)-countMap.get(b)
        );

        for(int key : countMap.keySet()){
            maxHeap.add(key);
            if(maxHeap.size()>k){
                maxHeap.poll();
            }
        }

        int[] ans = new int[k];

        for(int i=0;i<k;i++){
            ans[i] = maxHeap.poll();
        }

        return ans;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-4.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        di = dict()
        for i in range(len(nums)):
            di[nums[i]] = di.get(nums[i],0)+1
            if(di[nums[i]]>=2):
                return True
        return False
        